File: backend/app/shared/roles.py
"""System roles.

This enum is the single source of truth on the backend and must stay in sync
with ``frontend-v2/src/config/roles.ts``.
"""

from __future__ import annotations

from enum import StrEnum


class Role(StrEnum):
    """Role assigned to a user inside a tenant."""

    PROPIETARIO = "PROPIETARIO"
    ADMIN_OPERACIONES = "ADMIN_OPERACIONES"
    AGRONOMO = "AGRONOMO"
    OPERADOR_DRONE = "OPERADOR_DRONE"
    APLICADOR = "APLICADOR"
    BODEGUERO = "BODEGUERO"
    JEFE_BODEGA = "JEFE_BODEGA"
    AUDITOR = "AUDITOR"
    APICULTOR = "APICULTOR"


ROLE_LABELS: dict[Role, str] = {
    Role.PROPIETARIO: "Propietario",
    Role.ADMIN_OPERACIONES: "Admin Operaciones",
    Role.AGRONOMO: "Agrónomo",
    Role.OPERADOR_DRONE: "Operador de Drone",
    Role.APLICADOR: "Aplicador",
    Role.BODEGUERO: "Bodeguero",
    Role.JEFE_BODEGA: "Jefe de Bodega",
    Role.AUDITOR: "Auditor",
    Role.APICULTOR: "Apicultor Registrado",
}

ROLES_2FA_REQUIRED: frozenset[Role] = frozenset(
    {
        Role.PROPIETARIO,
        Role.ADMIN_OPERACIONES,
        Role.JEFE_BODEGA,
        Role.AUDITOR,
        Role.AGRONOMO,
    }
)

File: frontend/src/config/roles.ts
export const ROLES = {
  PROPIETARIO: 'PROPIETARIO',
  ADMIN_OPERACIONES: 'ADMIN_OPERACIONES',
  AGRONOMO: 'AGRONOMO',
  OPERADOR_DRONE: 'OPERADOR_DRONE',
  APLICADOR: 'APLICADOR',
  BODEGUERO: 'BODEGUERO',
  JEFE_BODEGA: 'JEFE_BODEGA',
  AUDITOR: 'AUDITOR',
  APICULTOR: 'APICULTOR',
} as const

export type Role = (typeof ROLES)[keyof typeof ROLES]

export const ROLE_LABELS: Record<Role, string> = {
  PROPIETARIO: 'Propietario',
  ADMIN_OPERACIONES: 'Admin Operaciones',
  AGRONOMO: 'Agrónomo',
  OPERADOR_DRONE: 'Operador de Drone',
  APLICADOR: 'Aplicador',
  BODEGUERO: 'Bodeguero',
  JEFE_BODEGA: 'Jefe de Bodega',
  AUDITOR: 'Auditor',
  APICULTOR: 'Apicultor Registrado',
}
